src/_pytest/pathlib.py

```diff
@@ -254,11 +254,10 @@ def maybe_delete_a_numbered_dir(path: Path) -> None:
     lock_path = None
     try:
         lock_path = create_cleanup_lock(path)
-        parent = path.parent
-
-        garbage = parent.joinpath(f"garbage-{uuid.uuid4()}")
-        path.rename(garbage)
-        rm_rf(garbage)
+        with tempfile.TemporaryDirectory(
```

Comment:
This should spec a root directory and a prefix

Reply:
dir is the root directory